File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Shared/ResponseJsons/CompanyCreateResponse.cs
using BeltmanSoftwareDesign.Shared.Jsons;

namespace BeltmanSoftwareDesign.Shared.ResponseJsons
{
    public class CompanyCreateResponse : Response
    {
        public Company? Company {  get; set; }
        public bool ErrorCompanyNameAlreadyUsed { get; set; }
    }
}

File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Angular/src/app/interfaces/response/companycreateresponse.ts
import { Company } from "../company";
import { State } from "../state";

export interface CompanyCreateResponse {
    company: Company | null;
    errorCompanyNameAlreadyUsed: boolean;
    success: boolean;
    errorAuthentication: boolean;
    errorItemNotFound: boolean;
    state: State | null;
}